src/Storages/Distributed/DistributedBlockOutputStream.cpp

```diff
@@ -90,6 +90,7 @@ DistributedBlockOutputStream::DistributedBlockOutputStream(
     UInt64 insert_timeout_)
     : context(context_)
     , storage(storage_)
+    , table_lock(storage.lockForShare(context.getCurrentQueryId(), context.getSettingsRef().lock_acquire_timeout))
```

Comment:
Seems like it's already locked [here](https://github.com/ClickHouse/ClickHouse/blob/5b482f41913e5fd5a6ca36854dc5ec66764347a6/src/DataStreams/PushingToViewsBlockOutputStream.cpp#L37), do we really need another lock holder?

Reply:
No, nice catch!